Misc/NEWS.d/next/Library/2024-03-19-19-42-25.gh-issue-116987.ZVKUH1.rst

```diff
@@ -0,0 +1 @@
+Remove regex pattern check in :func:`inspect.findsource` which makes the function work with class code objects.
```

Comment:
"Remove regex pattern check" is an implementation detail which users do not need to know.

The observable effect is that inspect.findsource now works with class code objects.